src/Api/PubnubApi/EventEngine/Presence/States/CooldownState.cs

```diff
@@ -0,0 +1,42 @@
+using PubnubApi;
+using PubnubApi.EventEngine.Presence.Invocations;
+using PubnubApi.EventEngine.Core;
+using System.Collections.Generic;
+using System;
+
+namespace PubnubApi.EventEngine.Presence.States
+{
+    public class CooldownState : APresenceState
+    {
+        // TODO: Dummy Invocation until we have real ones
+        public override IEnumerable<IEffectInvocation> OnEntry => new DummyInvocation().AsArray();
+        public override IEnumerable<IEffectInvocation> OnExit => new DummyInvocation().AsArray();
+
+        // TODO: transitions
+        public override TransitionResult Transition(IEvent ev)
+        {
+            return ev switch 
+            {
+                Events.JoinedEvent e => new HeartbeatingState()
+                {
+                    Input = e.Input != this.Input ? this.Input + e.Input : this.Input,
+                },
+                Events.LeftEvent e => HandleLeftEvent(e),
+                Events.LeftAllEvent e => new InactiveState(),
+                Events.HeartbeatSuccessEvent e => new CooldownState()
```

Comment:
Do we need to handle `HeratbeatSuccess` when we are on `Cooldown` state ?

Reply:
Good catch. Probably I've added that in favour of Heartbeating and Reconnecting! I will change that after the compiling fix.